packages/page-staking/src/Performance/useAbftScores.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright 2017-2025 @polkadot/app-staking authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { Option, Struct, Vec } from '@polkadot/types';
+import type { Hash, SessionIndex } from '@polkadot/types/interfaces';
+import type { u16, u32 } from '@polkadot/types-codec';
+
+import { useEffect, useState } from 'react';
+
+import { getCommitteeManagement } from '@polkadot/react-api';
+import { createNamedHook, useApi, useCall } from '@polkadot/react-hooks';
+
+interface Score extends Struct {
+  sessionId: SessionIndex,
+  nonce: u32,
+  points: Vec<u16>
+}
+
+const OPT_BOND = {
+  transform: (value: Option<Score>): Score | undefined =>
+    value.isSome
+      ? value.unwrap()
+      : undefined
+};
+
+function AbftScoresImpl (session: number) {
+  const { api } = useApi();
+
+  const [firstBlockSessionHash, setFirstBlockSessionHash] = useState<Hash | undefined>(undefined);
+  const [scoresEnabled, setScoresEnabled] = useState<boolean>(false);
+  const sessionPeriod = Number(getCommitteeManagement(api).consts.sessionPeriod.toString());
+
+  useEffect(() => {
+    api.rpc.chain.getBlockHash(sessionPeriod * session + 1)
+      .then((blockHash) => setFirstBlockSessionHash(blockHash))
+      .catch(console.error);
+  },
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  [api, session]
+  );
+
+  useEffect(() => {
+    if (firstBlockSessionHash) {
+      api.at(firstBlockSessionHash.toString()).then((apiAtFirstBlock) => {
+        if (apiAtFirstBlock.query.aleph.abftScores !== undefined) {
+          apiAtFirstBlock.query.aleph.finalityVersion()
+            .then((finalityVersion) =>
+              setScoresEnabled(Number(finalityVersion.toString()) >= 5))
+            .catch(console.error);
+        }
+      }).catch(console.error);
+    }
+  }, [api, firstBlockSessionHash]);
```

Comment:
Looks like these 2 `useEffects` are 2 steps of essentially one functionality, getting the `scoresEnabled` value? Then you could merge them into one `useEffect`, thanks to which: a). you wouldn't need the `firstBlockSessionHash` state, b). you wouldn't cause component rerender in the intermediate state between those 2 phases, which intermediate state is not relevant for the logic. So basically sth like:
```suggestion
  useEffect(() => {
    api.rpc.chain.getBlockHash(sessionPeriod * session + 1)
      .then((firstBlockchainHash) => {
        api.at(firstBlockSessionHash.toString()).then((apiAtFirstBlock) => {
          if (apiAtFirstBlock.query.aleph.abftScores !== undefined) {
            apiAtFirstBlock.query.aleph.finalityVersion()
              .then((finalityVersion) =>
                setScoresEnabled(Number(finalityVersion.toString()) >= 5))
              .catch(console.error);
          }
        })
      })
      .catch(console.error);
  }, [api, session]);
```

Reply:
You're right, again this was the copy-paste problem and the fact I did not pay attention to minimize re-renders. Now I will do, thanks!